app/services/gift_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module GiftService
+  module_function
+
+  def call
+    user_dates = []
+
+    User.where(state: 'incompleted').find_in_batches do |group|
+      group.each do |u|
+        last_date = u.receipt.max_by do |receipt|
+          Time.zone.parse(
+            receipt['github_pull_request']['graphql_hash']['createdAt']
+          )
+        end
+        date = last_date['github_pull_request']['graphql_hash']['createdAt']
+        score = u.receipt.count
+        user_dates << { id: u.id, score: score, date: date }
+      end
+    end
+
+    sorted = user_dates.sort do |a, b|
+      a_date = Time.zone.parse(a[:date])
+      b_date = Time.zone.parse(b[:date])
+
+      [b[:score], a_date] <=> [a[:score], b_date]
```

Comment:
This logic doesn't seem right. You're mixing `a` and `b` on both sides.

Reply:
See: https://docs.ruby-lang.org/en/2.0.0/Array.html#method-i-3C-3D-3E

> Arrays are compared in an “element-wise” manner; the first two elements that are not equal will determine the return value for the whole comparison.

```suggestion
      [a[:score], a_date] <=> [b[:score], b_date]
```

Reply:
The logic your suggestion gives us seems right, but I don't think it's what we want. It prioritizes first the dates of the lowest scoring objects and then does the same for the higher scoring objs.  Ultimately the score is not being prioritized.
![Screen Shot 2019-11-18 at 10 12 55 AM](https://user-images.githubusercontent.com/40510669/69064763-abf34480-09ec-11ea-8c4a-db15d5cb1580.png)

To achieve what we want - first sorting by score then by date, we need to mix the `a` and `b` variables on both sides. This seems to force the `score` to be the first priority when comparing elements.
[Screen Shot 2019-11-18 at 10 11 36 AM](https://user-images.githubusercontent.com/40510669/69065057-1906da00-09ed-11ea-9031-1b2cc9a115d8.png)